src/qt/createwalletdialog.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#include <qt/askpassphrasedialog.h>
+#include <qt/bitcoingui.h>
+#include <qt/createwalletdialog.h>
+#include <qt/forms/ui_createwalletdialog.h>
+#include <wallet/wallet.h>
+
+#include <QMessageBox>
+#include <QProgressDialog>
+#include <QString>
+
+CreateWalletDialog::CreateWalletDialog(BitcoinGUI* parent) :
+    QDialog(parent),
+    ui(new Ui::CreateWalletDialog),
+    m_parent(parent)
+{
+    ui->setupUi(this);
+}
+
+CreateWalletDialog::~CreateWalletDialog()
+{
+    delete ui;
+}
+
+void CreateWalletDialog::SetWalletController(WalletController* wallet_controller)
+{
+    this->m_wallet_controller = wallet_controller;
+}
+
+void CreateWalletDialog::accept()
+{
+    // Get the options
+    std::string wallet_name = ui->wallet_name_line_edit->text().toStdString();
+    bool disable_priv_keys = ui->disable_privkeys_checkbox->isChecked();
+    bool blank = ui->blank_wallet_checkbox->isChecked();
+    bool encrypt = ui->encrypt_wallet_checkbox->isChecked();
+
+    // Get wallet creation flags
+    uint64_t flags = 0;
+    if (disable_priv_keys) {
+        flags |= WALLET_FLAG_DISABLE_PRIVATE_KEYS;
+    }
+    if (blank || encrypt) {
+        flags |= WALLET_FLAG_BLANK_WALLET;
+    }
+
+    // Check that the wallet doesn't already exist
+    if (m_wallet_controller->checkWalletExists(wallet_name)) {
+        QMessageBox::critical(this, tr("Wallet creation failed"), tr("A wallet with the name <b>%1</b> already exists").arg(QString(wallet_name.c_str()).toHtmlEscaped()));
+        QDialog::reject();
+        return;
+    }
+
+    // Show a progress dialog
+    QProgressDialog* dialog = new QProgressDialog(this);
+    dialog->setLabelText(tr("Creating Wallet <b>%1</b>...").arg(QString(wallet_name.c_str()).toHtmlEscaped()));
+    dialog->setRange(0, 0);
+    dialog->setCancelButton(nullptr);
+    dialog->setWindowModality(Qt::ApplicationModal);
+    dialog->show();
+
+    // Create the wallet
+    std::unique_ptr<interfaces::Wallet> wallet = m_wallet_controller->createWallet(wallet_name, flags);
+
+    if (wallet) {
```

Comment:
This isn't great. If I create a wallet with 'Encrypt Wallet' set, but then 'Cancel' from the 'Encrypt Wallet' dialog, the wallet will be created but not encrypted. That shouldn't happen.

Reply:
Hmm. I think fixing that requires a significant change to how `AskPassphraseDialog` works due to that dialog being the one that also encrypts.

Reply:
I think it needs to be done. Creating an unencrypted wallet by accident when the user backs out of the change password  dialog is bad user experience.

Reply:
I'll look into it

Reply:
Done